File: samplejava/src/main/java/pyxis/uzuki/live/richutils/samplejava/MiscActivity.java
package pyxis.uzuki.live.richutils.samplejava;

import android.os.Bundle;
import android.support.annotation.Nullable;

import pyxis.uzuki.live.richutilskt.utils.RichUtils;

public class MiscActivity extends BaseActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        int versionCode = RichUtils.versionCode(this);
        String versionName = RichUtils.versionName(this);

        int dip15 = RichUtils.dip2px(this, 15);
        int sp12 = RichUtils.sp2px(this, 12);
        int dip15f = RichUtils.dip2px(this, 15f);
        int sp12f = RichUtils.sp2px(this, 12f);

        float px2dip15 = RichUtils.px2dip(this, 15);
        float px2sp12 = RichUtils.px2sp(this, 12);

        int networkState = RichUtils.checkNetwork(this);
        String hash = RichUtils.getKeyHash(this);
    }
}

File: sample/src/main/java/pyxis/uzuki/live/richutilssample/MiscActivity.kt
package pyxis.uzuki.live.richutilssample

import android.annotation.SuppressLint
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_misc.*
import pyxis.uzuki.live.richutilskt.utils.*

class MiscActivity : AppCompatActivity() {

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_misc)

        val versionCode = versionCode()
        val versionName = versionName()

        val dip15 = dip2px(15)
        val sp12: Int = sp2px(12)
        val dip15f = dip2px(15f)
        val sp12f = sp2px(12f)

        val px2dip15 = px2dip(15)
        val px2sp12 = px2sp(12)

        val networkState = checkNetwork()
        val hash = getKeyHash()

        txtResult.text = "versionCode $versionCode, versionName $versionName, dip15 $dip15, sp12 $sp12," +
                "dip15f $dip15f, sp12f $sp12f, px2dip15 $px2dip15, px2sp12 $px2sp12, networkState $networkState, hash $hash"

        applyUnReadCount(5)

    }

}
